From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Benjamin Marzinski <bmarzins@redhat.com>
Date: Tue, 17 Dec 2024 20:30:30 -0500
Subject: [PATCH] libmultipath: export udev pthread cleanup functions

A future patch will make use of cleanup_udev_enumerate_ptr() and
cleanup_udev_device_ptr().

Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 libmultipath/discovery.c          | 4 ++--
 libmultipath/discovery.h          | 2 ++
 libmultipath/libmultipath.version | 6 ++++++
 3 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/libmultipath/discovery.c b/libmultipath/discovery.c
index b24594cd..672c783b 100644
--- a/libmultipath/discovery.c
+++ b/libmultipath/discovery.c
@@ -146,7 +146,7 @@ path_discover (vector pathvec, struct config * conf,
 		return pathinfo(pp, conf, flag);
 }
 
-static void cleanup_udev_enumerate_ptr(void *arg)
+void cleanup_udev_enumerate_ptr(void *arg)
 {
 	struct udev_enumerate *ue;
 
@@ -157,7 +157,7 @@ static void cleanup_udev_enumerate_ptr(void *arg)
 		(void)udev_enumerate_unref(ue);
 }
 
-static void cleanup_udev_device_ptr(void *arg)
+void cleanup_udev_device_ptr(void *arg)
 {
 	struct udev_device *ud;
 
diff --git a/libmultipath/discovery.h b/libmultipath/discovery.h
index b6eea258..c2a88686 100644
--- a/libmultipath/discovery.h
+++ b/libmultipath/discovery.h
@@ -58,6 +58,8 @@ bool has_uid_fallback(struct path *pp);
 int get_uid(struct path * pp, int path_state, struct udev_device *udev,
 	    int allow_fallback);
 bool is_vpd_page_supported(int fd, int pg);
+void cleanup_udev_enumerate_ptr(void *arg);
+void cleanup_udev_device_ptr(void *arg);
 
 /*
  * discovery bitmask
diff --git a/libmultipath/libmultipath.version b/libmultipath/libmultipath.version
index 40d9246d..e2cce8c7 100644
--- a/libmultipath/libmultipath.version
+++ b/libmultipath/libmultipath.version
@@ -307,3 +307,9 @@ LIBMULTIPATH_9.1.3 {
 global:
 	partmap_in_use;
 } LIBMULTIPATH_9.1.2;
+
+LIBMULTIPATH_9.1.4 {
+global:
+	cleanup_udev_enumerate_ptr;
+	cleanup_udev_device_ptr;
+} LIBMULTIPATH_9.1.3;
